### ISSUE TITLE
symbol 'o' not supported.

### ISSUE THREAD

**Issue by hisea** [gaynetdinov/ex_marshal#13]
I'm using this lib to decode rails 3 sessions.

It seems that `ExMarshal` doesnt support ruby object which Rails 3.2 puts object in flash.

i.e.

in Ruby
```
cookie = "BAh7CUkiD3Nlc3Npb25faWQGOgZFVEkiJTRjODRkMzUzMTFkNTc2YWUwYjVkMmNjZjRhNjY4YzY2BjsAVEkiE3VzZXJfcmV0dXJuX3RvBjsAVCIGL0kiEF9jc3JmX3Rva2VuBjsARkkiMWVlQkRhOThqT2F2Q2dkTFRSemZkM2lpMTU4Ly9JckUxVEJrY1lwZVgwQnM9BjsARkkiCmZsYXNoBjsAVG86JUFjdGlvbkRpc3BhdGNoOjpGbGFzaDo6Rmxhc2hIYXNoCToKQHVzZWRvOghTZXQGOgpAaGFzaH0GOgphbGVydFRGOgxAY2xvc2VkRjoNQGZsYXNoZXN7BjsKSSI2WW91IG5lZWQgdG8gc2lnbiBpbiBvciBzaWduIHVwIGJlZm9yZSBjb250aW51aW5nLgY7AFQ6CUBub3cw" 

b= Base64.decode64(cookie)
=> "\x04\b{\tI\"\x0Fsession_id\x06:\x06ETI\"%4c84d35311d576ae0b5d2ccf4a668c66\x06;\x00TI\"\x13user_return_to\x06;\x00T\"\x06/I\"\x10_csrf_token\x06;\x00FI\"1eeBDa98jOavCgdLTRzfd3ii158//IrE1TBkcYpeX0Bs=\x06;\x00FI\"\nflash\x06;\x00To:%ActionDispatch::Flash::FlashHash\t:\n@usedo:\bSet\x06:\n@hash}\x06:\nalertTF:\f@closedF:\r@flashes{\x06;\nI\"6You need to sign in or sign up before continuing.\x06;\x00T:\t@now0"

Marshal.load b
=> {"session_id"=>"4c84d35311d576ae0b5d2ccf4a668c66",
 "user_return_to"=>"/",
 "_csrf_token"=>"eeBDa98jOavCgdLTRzfd3ii158//IrE1TBkcYpeX0Bs=",
 "flash"=>
  #<ActionDispatch::Flash::FlashHash:0x007fe33510dc70
   @closed=false,
   @flashes={:alert=>"You need to sign in or sign up before continuing."},
   @now=nil,
   @used=#<Set: {:alert}>>}
```

Where in ExMarshal I get:
```
** (ExMarshal.DecodeError) term which starts with the following symbol is not supported: "o"                                                                       lib/ex_marshal/decoder.ex:39: ExMarshal.Decoder.decode_element/2
               lib/ex_marshal/decoder.ex:244: ExMarshal.Decoder.decode_hash/4
               lib/ex_marshal/decoder.ex:3: ExMarshal.Decoder.decode/1
```

**Comment by hisea** [comment]
I can open a PR to ignore o tags and just fill them with blanks, this will works for me since I dont intent to read Ruby objects in Elixir anyway.

But wondering if this approach is good enough for everyone.

Thanks

**Comment by gaynetdinov** [comment]
Yeah, `ex_marshal` does not support decoding ruby objects and yes, encoded ruby objects start with `o` symbol in stream. 

I think the best approach would be to have a flag, which would indicate that `ex_marshal` should ignore ruby objects and put empty string or nil as a value. That way default behaviour of `ex_marshal` would be the same for everyone and only if you pass this flag, `ex_marshal` would change its behaviour. What do you think?

**Comment by hisea** [comment]
I think this is a great approach. I already have a branch that sets nil for ruby objects. I can work on adding the config option to enable nil objects.

**Comment by gaynetdinov** [comment]
Published v0.0.7 on hex.pm.
